### PR TITLE
 feat(template): replace openapiv2 generator with gnostic

### PR DESCRIPTION
**Pull request by Ricagraca**
This PR replaces the current openapiv2 generator for the oas with openapi from gnostic. The annotations for v3 were imported in the template proto file, instead of the v2 ones.

### CODE REVIEW

File: _template/api/proto/{{.Base.appName}}/v1/{{.Base.appName}}.proto [hunk 1]
```diff
@@ -9,34 +9,23 @@ import "google/protobuf/empty.proto";
...
 // and the package name.
 option go_package = "{{.Base.moduleName | replace `.git` ``}}/internal/pkg/pb;pb";
 
-{{- if .Extensions.grpc.grpcGateway }}
-// These annotations are used when generating the OpenAPI file.
-option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_swagger) = {
-  swagger: "2.0"
-  info: {
-    title: "{{.Base.appName}}-api"
-    version: "1.0"
```
Comment (linuxluigi) [inline _template/api/proto/{{.Base.appName}}/v1/{{.Base.appName}}.proto:25]: Please keep an example, ho to use it.

File: _template/api/proto/buf.yaml [hunk 2]
```diff
@@ -6,11 +6,12 @@ deps:
...
-  - buf.build/grpc-ecosystem/grpc-gateway
+  - buf.build/gnostic/gnostic
 {{- end }}
 lint:
   use:
     - DEFAULT
   except:
     - RPC_RESPONSE_STANDARD_NAME
     - RPC_REQUEST_RESPONSE_UNIQUE
+    - IMPORT_NO_WEAK # required by gnostic for now. this way gen-go will not import _ "" broken packages
```
Comment (linuxluigi) [inline _template/api/proto/buf.yaml:17]: As I understood it, it is only used when gRPC Gateway is in use, so could you please only import it when gRPC Gateway was enabled:

```yaml
{{- if .Extensions.grpc.grpcGateway }}
    - IMPORT_NO_WEAK # required by gnostic for now. this way gen-go will not import _ "" broken packages
{{- end }}
```

Comment (Ricagraca) [reply _template/api/proto/buf.yaml:17]: You are right!

**Comment by linuxluigi** [comment]
Looks good to me, will try to find some time today to test your code

File: embed_gen.go [hunk 3]
```diff
@@ -3,5 +3,5 @@ package gotemplate
 
 import "embed"
 
-//go:embed _template _template/.azure-pipelines.yml _template/.dockerignore _template/.editorconfig _template/.githooks _template/.github _template/.gitignore _template/.gitlab-ci.yml _template/.golangci.yml
+//go:embed _template _template/.azure-pipelines.yml _template/.dockerignore _template/.editorconfig _template/.githooks _template/.github _template/.gitignore _template/.gitlab-ci.yml _template/.golangci.yml _template/.vscode
```
Comment (brumhard) [inline embed_gen.go:6]: Could you remove the .vscode folder here again pls?. I think it landed there accidentally on make generate right?
We should probably exclude stuff that is in gitignore during generation, I will note that 👀

File: _template/api/proto/buf.yaml [hunk 4]
```diff
@@ -6,11 +6,14 @@ deps:
...
+  - buf.build/gnostic/gnostic
 {{- end }}
 lint:
   use:
     - DEFAULT
   except:
     - RPC_RESPONSE_STANDARD_NAME
     - RPC_REQUEST_RESPONSE_UNIQUE
+{{- if .Extensions.grpc.grpcGateway }}
+    - IMPORT_NO_WEAK # required by gnostic for now. this way gen-go will not import _ "" broken packages
```
Comment (brumhard) [inline _template/api/proto/buf.yaml:18]: Is it needed to exclude that globally if you also add the `// buf:lint:ignore IMPORT_NO_WEAK` directive in the proto file?

File: _template/api/proto/{{.Base.appName}}/v1/{{.Base.appName}}.proto [hunk 5]
```diff
@@ -18,15 +21,31 @@ option go_package = "{{.Base.moduleName | replace `.git` ``}}/internal/pkg/pb;pb
 
 {{- if .Extensions.grpc.grpcGateway }}
 // These annotations are used when generating the OpenAPI file.
-option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_swagger) = {
-  swagger: "2.0"
+option (openapi.v3.document) = {
```
Comment (brumhard) [inline _template/api/proto/{{.Base.appName}}/v1/{{.Base.appName}}.proto:24]:
```suggestion
option (gnostic.openapi.v3.document) = {
```

**Review by brumhard — CHANGES_REQUESTED** [review]
Hi there,
sry for keeping you waiting for so long. I really appreciate the effort. In fact, I've been looking for a way to create openapiv3 specs for a long time so this is a great addition 👍🏻 

I found a few minor things while testing.
Also since it's not included in the PR: could you also fix `_template/buf.gen.yaml` and rename
```
  - name: openapiv2
    out: api/openapi
```

to 
```
  - name: openapi
    out: api/openapi
```

Thank you very much :)

**Review by linuxluigi — APPROVED** [review]

**Review by brumhard — APPROVED** [review]
lgtm

**Pull request merged**